### PR TITLE
populate the app ID in the FAN timeout notif url with the publisher ID and the auction with the request ID

### PR DESCRIPTION
**Pull request by Aadeshp**

### CODE REVIEW

File: adapters/audienceNetwork/facebook.go [hunk 1]
```diff
@@ -455,18 +460,39 @@ func NewFacebookBidder(client *http.Client, platformID string, appSecret string)
...
 	if err != nil {
 		return &adapters.RequestData{}, []error{err}
 	}
 
-	uri := fmt.Sprintf("https://www.facebook.com/audiencenetwork/nurl/?partner=%s&app=%s&auction=%s&ortb_loss_code=2", fa.platformID, fa.platformID, auction_id)
+	// The publisher ID is either in the app object or the site object, depending on the supply of the request so we need
+	// to check both
+	pubID, err = jsonparser.GetString(req.Body, "app", "publisher", "id")
+	if err != nil {
+		pubID, err = jsonparser.GetString(req.Body, "site", "publisher", "id")
```
Comment (SyntaxNode) [inline adapters/audienceNetwork/facebook.go:481]: What if the app.publisher.id exists and is empty?

File: adapters/audienceNetwork/facebook_test.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func TestJsonSamples(t *testing.T) {
 
 func TestMakeTimeoutNotice(t *testing.T) {
 	req := adapters.RequestData{
```
Comment (SyntaxNode) [inline adapters/audienceNetwork/facebook_test.go:47]: Please add a unit test for getting the pubID from the site.publisher.id.

File: adapters/audienceNetwork/facebook.go (hunk 1 above)
Comment (Aadeshp) [reply adapters/audienceNetwork/facebook.go:481]: we would have rejected the impression — we validate that there is a publisher ID before sending a request to FAN (in extractPlacementAndPublisher)

Comment (hhhjort) [reply adapters/audienceNetwork/facebook.go:481]: We need to consider the case where the request comes from somewhere other than Appnexus.

Comment (SyntaxNode) [reply adapters/audienceNetwork/facebook.go:481]: The validation @Aadeshp mentions isn't specific to AppNexus, is it? The `extractPlacementAndPublisher` method is in the PBS adapter.

Comment (hhhjort) [reply adapters/audienceNetwork/facebook.go:481]: Oh, yes, the validation is in the facebook adapter. So it should be safe.

**Pull request merged**